### PR TITLE
[NanoAOD] Fix typos in jetsAK4 CHS and Puppi tables

### PR DESCRIPTION
**Pull request by hqucms**
#### PR description:

This pull request fixes a few typos in the jetsAK4 CHS and Puppi tables:

(1) Fix to the `btagRobustParTAK4CvB` definition: the previous definition was using the `pfDeepFlavourJetTags` instead of pfParticleTransformerAK4JetTags` for the `probbb` and `problepb` scores in the denominator. This affects both the CHS and Puppi AK4 jet tables, and **probably(?) affects the official production.**

(2) Fix to the `nanoAOD_addDeepInfoAK4()` fucntion in the `jetsAK4_Puppi_cff.py`: *this should only affects private/custom productions.*

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by simonepigazzini** [comment]
@hqucms can you elaborate on the fix? I think this affects only custom production in which either PNet or PartT are evaluated on the fly while producing NANO, right?

Thank you

**Comment by hqucms** [comment]
> @hqucms can you elaborate on the fix? I think this affects only custom production in which either PNet or PartT are evaluated on the fly while producing NANO, right?
> 
> Thank you

Hi @simonepigazzini -- yes that's correct.

**Comment by simonepigazzini** [comment]
@cms-sw/btv-pog-l2 please have a look.

**Comment by simonepigazzini** [comment]
enable nano

**Comment by simonepigazzini** [comment]
please test

**Comment by simonepigazzini** [comment]
please test

File: PhysicsTools/NanoAOD/python/custom_jme_cff.py [hunk 1]
```diff
@@ -182,13 +182,13 @@
...
-  btagRobustParTAK4QG  = Var("?(pt>=15)&&(bDiscriminator('pfParticleTransformerAK4JetTags:probg')+bDiscriminator('pfParticleTransformerAK4JetTags:probuds'))>0?bDiscriminator('pfParticleTransformerAK4JetTags:probg')/(bDiscriminator('pfParticleTransformerAK4JetTags:probg')+bDiscriminator('pfParticleTransformerAK4JetTags:probuds')):-1",float,doc="DeepJet g vs uds discriminator",precision=10),
+  btagRobustParTAK4B   = Var("?(pt>=15)?bDiscriminator('pfParticleTransformerAK4JetTags:probb')+bDiscriminator('pfParticleTransformerAK4JetTags:probbb')+bDiscriminator('pfParticleTransformerAK4JetTags:problepb'):-1",float,doc="RobustParTAK4 b+bb+lepb tag discriminator",precision=10),
+  btagRobustParTAK4C   = Var("?(pt>=15)?bDiscriminator('pfParticleTransformerAK4JetTags:probc'):-1",float,doc="RobustParTAK4 charm tag raw score",precision=10),
```
[the remark this answers is not in the document]
Comment (simonepigazzini) [reply PhysicsTools/NanoAOD/python/custom_jme_cff.py:186]: @swertz confirmed that this needs a fix for the ongoing production

**Comment by simonepigazzini** [comment]
@hqucms do you expect more changes?

@cms-sw/btv-pog-l2 @swertz, in the [nanoDQM](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_13_3_X_2023-08-29-2300+6e83d2/58655/dqm-histo-comparison-summary.html) the test workflows we can see a change in the CvsB output node. Can you have a look and comment on the change? Do you have slides from your own comparison of the DeepJet and RobustPartT for that node?

Thank you

**Comment by hqucms** [comment]
> @hqucms do you expect more changes?

Not from my side.

**Comment by simonepigazzini** [comment]
+1

@hqucms can you open a backport to 13_0 please?

File: PhysicsTools/NanoAOD/python/custom_jme_cff.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rappoccio) [reply PhysicsTools/NanoAOD/python/custom_jme_cff.py:186]: Do we have a plan? Are we stopping the production?

[the remark this answers is not in the document]
Comment (simonepigazzini) [reply PhysicsTools/NanoAOD/python/custom_jme_cff.py:186]: The production is stopped: https://its.cern.ch/jira/browse/PDMVRERECO-72?focusedId=4926345&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4926345.

The plan is to backport this PR to 13_0 asap and restart the production.

**Comment by hqucms** [comment]
> +1
> 
> @hqucms can you open a backport to 13_0 please?

Hi @simonepigazzini -- sure I will do it today.

**Comment by hqucms** [comment]
Backport to 13_0_X created: https://github.com/cms-sw/cmssw/pull/42710

**Comment by antoniovilela** [comment]
+1

**Pull request merged**